hydroflow_lang/src/graph/ops/persist.rs

```diff
@@ -46,6 +46,10 @@ pub const PERSIST: OperatorConstraints = OperatorConstraints {
     persistence_args: RANGE_0,
     type_args: RANGE_0,
     is_external_input: false,
+    // If this is set to true, the state will need to be cleared using `#context.set_state_tick_hook`
+    // to prevent reading uncleared data if this subgraph doesn't run.
+    // https://github.com/hydro-project/hydroflow/issues/1298
+    // If `'tick` lifetimes are added.
```

Comment:
Are we using the singleton feature as a proxy signal to indicate that the state needs to be clear for a tick?

For example, could there ever be a non-singleton-related state that may need clearing?

Reply:
Yeah. Other `'tick` state will need to be cleared, but existing implementation does that when the iterator runs, which is good enough. There is only a problem if a singleton can reference the state before the iterator runs, in that case.